src/webview/mod.rs

```diff
@@ -126,6 +126,19 @@ pub struct WebViewAttributes {
   /// allow to nivagate and false is not.
   pub navigation_handler: Option<Box<dyn Fn(String) -> bool>>,
 
+  /// Set a download handlers to manage incoming downloads.
+  ///
+  /// The first closure takes two parameters - the first is a `String` representing the url being downloaded from and and the 
+  /// second is a mutable `String` reference that (possibly) represents where the file will be downloaded to. The latter 
+  /// parameter can be used to set the download location by assigning a new path string to it - the assigned path _must_ be 
+  /// absolute, and (on Windows) cannot include a UNC prefix. The closure returns a `bool` to allow or deny the download.
+  /// The second closure is fired when the download completes, with a `String` representing the path to where the download was
+  /// saved and a `bool` indicating if the download succeeded.
+  pub download_handlers: Option<(
+    Box<dyn FnMut(String, &mut String) -> bool>,
```

Comment:
Probably better to use the `Path` type
```suggestion
    Box<dyn FnMut(String, &mut Path) -> bool>,
```

Reply:
Switched to using `&mut PathBuf` - `Path` size can't be known at compile time so we can't take a mutable reference.

Reply:
Yeah that's fine too.